### PR TITLE
Add decimals to fix issue with Paraswap

### PR DESCRIPTION
**Pull request by anxolin**
Paraswap price is failing for some tokens

## Context: 
https://gnosisinc.slack.com/archives/C0203Q9R4JJ/p1625039735002400

I just notice, we get an error with this endpoint:
https://apiv4.paraswap.io/v2/prices/?from=0x6DEA81C8171D0bA574754EF6F8b412F2Ed88c54D&[…]083C756Cc2&amount=4668313438455293225856&side=SELL&network=1

However, if you specify the decimals, it works:
https://apiv4.paraswap.io/v2/prices/?from=0x6DEA81C8171D0bA574754EF6F8b412F2Ed88c54D&[…]0000000000&fromDecimals=18&toDecimals=18&side=SELL&network=1

I was kind of assuming the decimals were optional if they are readable in the contract ... but is not the case!
https://etherscan.io/address/0x6DEA81C8171D0bA574754EF6F8b412F2Ed88c54D#readContract

## Screenshot

image

## Test
To see the problem, use LQTY in in Staging. It will only return GP price and it will fail for Paraswap. Test:
- Add LQTY by address `0x6DEA81C8171D0bA574754EF6F8b412F2Ed88c54D `
- Get an estimation
- Notice the error in the console, and the estimation is worse than Paraswap

TheIn this PR should work!
* Repeat

More tests. Maybe, try weird tokens?

### CODE REVIEW

File: src/custom/state/price/updater.ts [hunk 1]
```diff
@@ -11,6 +11,7 @@ import { QuoteInformationObject } from './reducer'
 import { useIsUnsupportedTokenGp } from 'state/lists/hooks/hooksMod'
 import useDebounceWithForceUpdate from 'hooks/useDebounceWithForceUpdate'
 import useIsOnline from 'hooks/useIsOnline'
+import { DEFAULT_DECIMALS } from '@src/custom/constants'
```
Comment (W3stside) [inline src/custom/state/price/updater.ts:14]: path

**Review by W3stside — APPROVED** [review]

File: src/custom/hooks/useRefetchPriceCallback.ts [hunk 2]
```diff
@@ -123,15 +124,21 @@ async function _getBestPriceQuote(params: PriceQuoteParams): Promise<PriceInform
...
   const feePromise =
     fetchFee || !previousFee
-      ? getFeeQuote({ chainId, sellToken, buyToken, amount, kind })
+      ? getFeeQuote({ chainId, sellToken, buyToken, fromDecimals, toDecimals, amount, kind })
       : Promise.resolve(previousFee)
 
+  // Log fee for debugging
+  feePromise.then(fee => {
+    console.log(`Fee: ${formatAtoms(fee.amount, fromDecimals)} (in atoms ${fee.amount})`)
+    return fee
```
Comment (alfetopito) [inline src/custom/hooks/useRefetchPriceCallback.ts:139]: nit: if this is just for debugging, no need to return it

File: src/custom/utils/format.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import BigNumber from 'bignumber.js'
+
+const TEN = new BigNumber(10)
+
+export function formatAtoms(amount: string, decimals: number): string {
+  return new BigNumber(amount).div(TEN.pow(decimals)).toString(10)
+}
```
Comment (alfetopito) [inline src/custom/utils/format.ts:7]: Sure there's no helper function to do that already?
I could swear I saw this somewhere...

**Comment by W3stside** [comment]
@alfetopito is right, you can use `formatUnits` and `parseUnits` from ethers

File: src/custom/state/price/updater.ts (hunk 1 above)
Comment (anxolin) [reply src/custom/state/price/updater.ts]: https://github.com/gnosis/cowswap/pull/847

**Pull request merged**